hyperjumptech/monika
# feature-456 format slack message using block kit

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Customise Slack message look and feel. Resolves #456 

## How did you implement / how did you fix it  
1.  Use Slack Block Kit

## How to test  
1. Create a new file called monika.yml with the following content
```
notifications:
  - id: unique-id-slack
    type: slack
    data:
      url: YOUR_SLACK_INCOMING_WEBHOOK_URL
probes:
  - id: sample_1
    name: Sample Request
    requests:
      - method: GET
        url: https://reqres.in/api/users/23
    alerts:
      - query: response.time > 200
        message: Response time is {{ response.time }} ms, expecting less than 200ms
      - query: response.status != 200
        message: Status code is not 200. I think their service is down.
      - query: isEmpty(response.body)
        message: Response body should not be empty!
    incidentThreshold: 2
    recoveryThreshold: 2
```

2. Run `npm start`
3. Start receiving notification


## Screenshots

**Pop-up notification on macOS**
<img width="370" alt="monika-456-start-notification" src="https://user-images.githubusercontent.com/5510786/137276781-bf1766fd-92b5-4111-b63e-c9374e1139a4.png">

**Slack messages**
<img width="699" alt="monika-456" src="https://user-images.githubusercontent.com/5510786/137276876-74b1ff23-7272-45ce-a536-0ed28dca4d7b.png">
